### PR TITLE
Oidc configuration endpoints

### PR DESCRIPTION
**Pull request by eguzki**
* Get OIDC configuration endpoint
* Update OIDC configuration endpoint

### CODE REVIEW

File: client/oidc.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	"fmt"
+	"net/http"
+)
+
+const (
+	oidcResourceEndpoint = "/admin/api/services/%d/proxy/oidc_configuration.json"
+)
+
+// OidcConfiguration fetches 3scale product oidc configuration
+func (c *ThreeScaleClient) OidcConfiguration(productID int64) (*OIDCConfiguration, error) {
```
Comment (miguelsorianod) [inline client/oidc.go:15]: being an acronym probably a good idea to call it `OIDCConfiguration` (same with the other methods/places where the acronym appears)

File: client/types.go [hunk 2]
```diff
@@ -635,3 +635,17 @@ type PolicyConfig struct {
 type PoliciesConfigList struct {
 	Policies []PolicyConfig `json:"policies_config"`
 }
+
+// OIDCConfigurationItem - Holds an OIDC configuration item object
+type OIDCConfigurationItem struct {
+	ID                        int64 `json:"id,omitempty"`
```
Comment (miguelsorianod) [inline client/types.go:641]: Is there some reason ID has omitempty struct tag and the others don't?

File: client/policies.go [hunk 3]
```diff
@@ -20,9 +19,6 @@ func (c *ThreeScaleClient) Policies(productID int64) (*PoliciesConfigList, error
 		return nil, err
 	}
 
-	urlValues := url.Values{}
```
Comment (miguelsorianod) [inline client/policies.go:23]: What's the context of this change?

Comment (miguelsorianod) [reply client/policies.go:23]: I see, it's in another commit. Not related to this PR but a small change

File: client/types.go (hunk 2 above)
Comment (eguzki) [reply client/types.go:641]: When reading OIDCConfiguration from 3scale API, the ID will be set. 
When updating OIDCConfiguration of your product, the serialized object is that one and `ID` field is not required to be sent. It not `omitempty`, the request would  send a `ID: 0` field which is unwanted.

Comment (miguelsorianod) [reply client/types.go:641]: I see. :+1:

**Pull request merged**